clippy_lints/src/ptr.rs

```diff
@@ -591,6 +587,7 @@ fn check_ptr_arg_usage<'tcx>(cx: &LateContext<'tcx>, body: &Body<'tcx>, args: &[
             };
             let args = &self.args[args_idx];
             let result = &mut self.results[args_idx];
+            let deref_ty = LazyCell::new(|| args.deref_ty.ty(self.cx));
```

Comment:
Do we need to use `LazyCell` here? I don't see why we can't just move this assignment down to where it's used considering there's only one path it's used

Reply:
Good point. The changes I made went through a bunch of iterations where this made more sense before I created the PR, but now it doesn't anymore, so removing